nexus/src/db/datastore/fleet.rs

```diff
@@ -0,0 +1,69 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! [`DataStore`] methods on [`Fleet`]s.
+
+use super::DataStore;
+use crate::authz;
+use crate::context::OpContext;
+use crate::db;
+use crate::db::error::public_error_from_diesel_pool;
+use crate::db::error::ErrorHandler;
+use crate::db::identity::Asset;
+use crate::db::model::Fleet;
+use crate::db::pagination::paginated;
+use async_bb8_diesel::AsyncRunQueryDsl;
+use diesel::prelude::*;
+use diesel::upsert::excluded;
+use omicron_common::api::external::DataPageParams;
+use omicron_common::api::external::Error;
+use omicron_common::api::external::ListResultVec;
+use omicron_common::api::external::ResourceType;
+use uuid::Uuid;
+
+impl DataStore {
+    pub async fn fleet_list(
+        &self,
+        opctx: &OpContext,
+        pagparams: &DataPageParams<'_, Uuid>,
+    ) -> ListResultVec<Fleet> {
+        opctx.authorize(authz::Action::Read, &authz::FLEET).await?;
+        use db::schema::fleet::dsl;
+        paginated(dsl::fleet, dsl::id, pagparams)
+            .select(Fleet::as_select())
+            .load_async(self.pool_authorized(opctx).await?)
+            .await
+            .map_err(|e| public_error_from_diesel_pool(e, ErrorHandler::Server))
+    }
+
+    /// Stores a new fleet in the database.
+    ///
+    /// This function is a no-op if the fleet already exists.
+    pub async fn fleet_insert(
```

Comment:
This behavior makes sense because it's being called from a populator, but it might be surprising because I think most of the "create" functions _don't_ work this way.  I think most of the other populators use `load_X` instead to suggest that they're not intended for general use.  I see `rack_insert` already exists so 🤷 I guess.  It just makes me nervous because if people misunderstand the behavior of this case, the impact can be significant and the bug can be very hard to catch.  (That happened before with #985.)

Reply:
I tried to document the behavior in the comment, indicating the no-op case on prior existence. Do you think there's a naming convention I should use to make this more clear?

(To me, `load_*` makes it sound like a read operation, not a write operation?)

Reply:
This function looks like the same pattern as the existing `load_*` functions and I think it'd be good to use a common naming convention to communicate this behavior.  I like `load_*` (as in: loading data into the database, which feels like a one-time initial setup operation) but I'm not attached to it.  We could change them all to something else -- `populate_*`?  `insert_fixed_*`?  `insert_initial_*`?

Reply:
I'm no longer inserting a record for the fleet, so this is no longer relevant.